### PR TITLE
Restructure terraform scripts

### PR DESCRIPTION
**Pull request by yhwang**
In order to support different IKS set, the terraform scripts/modules
under `terraform` directory are reorgnized to a better structure.
- all modules are moved to `terraform/modules`
- create `iks-classic` for classic cluster
- create `iks-vpc-gen2` for vpc gen2 cluster

### CODE REVIEW

File: terraform/kubeflow-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: networking.istio.io/v1alpha3
```
Comment (chinhuang007) [inline terraform/kubeflow-gateway.yaml:1]: Successfully deployed to classic and VPC clusters. So everything looks good. One question here for my understanding, where is this file used? And why there is a fixed host name?

**Comment by chinhuang007** [comment]
Failed to run the e2e mnist pipeline at the katib experiment step in a classic cluster, getting

HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Internal error occurred: failed calling webhook \"mutating.experiment.katib.kubeflow.org\": Post \"https://katib-controller.kubeflow.svc:443/mutate-experiments?timeout=30s\": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0","reason":"InternalError","details":{"causes":[{"message":"failed calling webhook \"mutating.experiment.katib.kubeflow.org\": Post \"https://katib-controller.kubeflow.svc:443/mutate-experiments?timeout=30s\": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0"}]},"code":500}https://katib-controller.kubeflow.svc:443/mutate-experiments?timeout=30s\": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0"}]},"code":500}

Not sure this has anything to do with the automation scripts or just an environment issue.

File: terraform/iks-classic/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+|  org                   | IBM Cloud organization | string  |  ✓ |  |  |
+|  space                 | The space in IBM Cloud | string  | ✓ |  |  |
+|  resource_group        | The resource group in IBM Cloud | string  |  | default |  |
+|  kube_version          | kubernetes version | string  |  | 1.19.9 |  |
+|  appid_plan            | The plan for AppID | string  |  | lite |  |
+|  appid_name            | Instnace name for AppID service | string  |   | appid-instance |  |
+|  cluster_name          | Cluster name | string  |   | my-kfp-cluster | |
+|  cluster_worker_flavor | The machine flavor for the cluster worker nodes | string  |   | b3c.4x16 |  |
+|  cluster_worker_count  | number of workers in the cluster | number  |   | 2 |  |
+|  zone                  | The zone where the cluster will be created | string  |   | dal10 | |
```
Comment (chinhuang007) [inline terraform/iks-classic/README.md:29]: Does it support multi-zone clusters?

File: terraform/kubeflow-gateway.yaml (hunk 1 above)
Comment (yhwang) [reply terraform/kubeflow-gateway.yaml:1]: it's not used now. originally, I want to use this in the ansible playbook. I embedded it into the ansible playbook. I can remove it.

File: terraform/iks-classic/README.md (hunk 2 above)
Comment (yhwang) [reply terraform/iks-classic/README.md:29]: nope, for multi-zone, need more customization for the vpc network

**Review by chinhuang007 — APPROVED** [review]
LGTM!

**Pull request merged**